### ISSUE TITLE
Finetuning error

### ISSUE THREAD

**Issue by Rongjiehuang** [AlanBaade/MAE-AST-Public#2]
Dear author,
Thanks for releasing your code! I have come across an issue when fine-tuning the model with provided checkpoints in the s3prl benchmark tool:

```
File "/home1/huangrongjie/Project/s3prl/s3prl/upstream/interfaces.py", line 250, in tolist
    assert abs(paired_feature.size(1) - round(max([len(wav) for wav in paired_wavs]) / self.downsample_rate)) < TOLERABLE_SEQLEN_DIFF
AssertionError
```

This error is reported with command `-k chunk_patch_75_12LayerEncoder.pt`, while disappears with `-k random_frame_75_12LayerEncoder.pt`.

Could you check up this code? Thanks.

**Comment by AlanBaade** [comment]
Thanks for the comment

I've run into this error in the past, but I didn't have a good way to address it in portable code (and forgot to). Thanks for letting me remember it. The solution is to comment out the assertion or add a small constant to TOLERABLE_SEQLEN_DIFF, here's why:

The cause is that when converting audio into 16x16 patches, it will always be the case that a patching strategy that uses all audio can create too many or too few tokens according to this assertion by s3prl, but only by a small constant amount. This is best seen by example: When using a time dimension of 3 frames, two 128 f x 2t tokens will be generated by our loader, one that gets the first two time frames, and one that gets the last time frame plus zero padding. However, if we choose to include all audio input data, when using patches, we still have to generate 8 patches, one for each frequency band, even though each patch has 13 zerod out frames. This puts us over TOLERABLE_SEQLEN_DIFF.

It doesn't matter that the patch model fails this assertion because this assertion is used to help time-frame classification tasks function (like CTC loss for ASR). The chunk model really only makes sense to evaluate on sequence classification because chunks overlap on time and have too wide time bands for useful classification.

To the best of my knowledge, this is the only hack I needed to change in the s3prl library (which isn't meant for patch based models) to allow fine-tuning.
